scripts/create_tflm_arduino.py

```diff
@@ -0,0 +1,504 @@
+    args: a string containing all the script arguments
+    is_dry_run: if True, do not execute any commands
+  """
+  cmd = f"python3 {str(path_to_script)} {args}"
+  print(f"Running command: {cmd}")
+  if not is_dry_run:
+    _ = subprocess.run(cmd,
+                       shell=True,
+                       check=True,
+                       stdout=subprocess.PIPE,
+                       stderr=subprocess.PIPE)
+
+
+def _create_directories(paths: List[Path], is_dry_run: bool = True) -> None:
+  """
+  Create directory tree(s) given list of pathnames
+
+  Args:
+    paths: A list of Path objects
+    is_dry_run: if True, do not execute any commands
+  """
+  for dir_path in paths:
+    print(f"Creating directory tree {str(dir_path)}")
+    if not dir_path.is_dir() and not is_dry_run:
+      dir_path.mkdir(mode=0o755, parents=True, exist_ok=True)
+
+
+def _copy_files(paths: Iterable[Tuple[Path, Path]],
+                is_dry_run: bool = True) -> None:
+  """
+  Copy files given list of source and destination Path tuples
+
+  Args:
+    paths: A list of tuples of Path objects.
+    Each tuple is of the form (source, destination)
+    is_dry_run: if True, do not execute any commands
+  """
+  #dir_path: Tuple[Path, Path]
+  for from_path, to_path in paths:
+    print(f"Copying {str(from_path)} to {str(to_path)}")
+    if not is_dry_run:
+      shutil.copy2(from_path, to_path)
+
+
+class ArduinoProjectGenerator:
+  """
+  Generate the TFLM Arduino library ZIP file
+  """
+
+  #
+  # private enums
+  #
+
+  @unique
+  class Manifest(Enum):
+    ADD = "Add",
+    REMOVE = "Remove",
+    SPECIAL_REPO = "Special Repo",
+    SPECIAL_BASE = "Special Base"
+    PATCH_SED = "Patch Sed"
+
+  #
+  # private methods
+  #
+
+  def __init__(self) -> None:
+    args = self._ParseArguments().parse_args()
+    self.base_dir = Path(args.base_dir)
+    if args.output_dir is None:
+      self.output_dir = Path(tempfile.gettempdir()) / "tflm_arduino"
+    else:
+      self.output_dir = Path(args.output_dir)
+    self.is_dry_run: bool = args.is_dry_run
+    if args.manifest_file is None:
+      self.manifest_path = Path("scripts/MANIFEST.ini")
+    else:
+      self.manifest_path = Path(args.manifest_file)
+
+    # generate list of examples by inspecting base_dir/examples
+    self.examples: List[str] = []
+    for path in self.base_dir.glob("examples/*"):
+      if path.is_dir:
+        self.examples.append(path.name)
+
+    # parse manifest file
+    manifest = self._ParseManifest()
+    self.add_list: List[Path] = manifest[self.Manifest.ADD]
+    self.remove_list: List[Path] = manifest[self.Manifest.REMOVE]
+    self.special_repo_list: List[Tuple[Path, Path]] = manifest[
+        self.Manifest.SPECIAL_REPO]
+    self.special_base_list: List[Tuple[Path, Path]] = manifest[
+        self.Manifest.SPECIAL_BASE]
+    self.patch_sed_list: List[Tuple[List[Path], List[str]]] = manifest[
+        self.Manifest.PATCH_SED]
```

Comment:
why not assign self.add_list etc. in the _parse_manifest function? Then we won't need to create and think about the intermediate dict struct.